jans-cedarling/cedarling/src/authz/mod.rs

```diff
@@ -132,45 +141,95 @@ impl Authz {
 
         // Convert [`AuthorizeEntitiesData`] to  [`cedar_policy::Entities`] structure,
         // hold all entities that will be used on authorize check.
-        let entities = entities_data.entities(Some(&schema.schema))?;
-
-        // Check authorize where principal is `"Jans::Workload"` from cedar-policy schema.
-        let workload_result: Option<Response> = if self.config.authorization.use_workload_principal
-        {
-            match self.execute_authorize(ExecuteAuthorizeParameters {
-                entities: &entities,
-                principal: principal_workload_uid.clone(),
-                action: action.clone(),
-                resource: resource_uid.clone(),
-                context: context.clone(),
-            }) {
-                Ok(resp) => Some(resp),
-                Err(err) => return Err(AuthorizeError::CreateRequestWorkloadEntity(err)),
-            }
-        } else {
-            None
-        };
+        let entities = entities_data.clone().entities(Some(&schema.schema))?;
+
+        let (workload_authz_result, workload_authz_info, workload_entity_claims) =
+            if let Some(workload) = entities_data.workload {
+                let principal = workload.uid();
+
```

Comment:
adding 2 lines we can avoid clone whole structure of `entities_data`

```suggestion
        let workload_principal = entities_data.workload.as_ref().map(|e| e.uid()).to_owned();
        let user_principal = entities_data.user.as_ref().map(|e| e.uid()).to_owned();

        // Convert [`AuthorizeEntitiesData`] to  [`cedar_policy::Entities`] structure,
        // hold all entities that will be used on authorize check.
        let entities = entities_data.entities(Some(&schema.schema))?;

        let (workload_authz_result, workload_authz_info, workload_entity_claims) =
            if let Some(workload) = workload_principal {
                let principal = workload;
```